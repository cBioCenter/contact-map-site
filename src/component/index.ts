export * from './ErrorMessageComponent';
export * from './FolderUploadComponent';
