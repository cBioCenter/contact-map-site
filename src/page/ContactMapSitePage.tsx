import { ConnectedRouter } from 'connected-react-router';
import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Container } from 'semantic-ui-react';

import { LandingPage, ManuscriptPage } from '~contact-map-site~/page';
import { history } from '~contact-map-site~/reducer';

export class ContactMapSitePage extends React.Component {
  public render() {
    return (
      <ConnectedRouter history={history}>
        <Container id={'ContactMapSitePage'} fluid={true}>
          <Switch>
            <Route path={'/manuscript'} render={this.renderManuscriptPage} />
            <Route exact={true} path={'/'} render={this.renderLandingPage} />
          </Switch>
        </Container>
      </ConnectedRouter>
    );
  }

  protected renderLandingPage = () => {
    return <LandingPage />;
  };

  protected renderManuscriptPage = () => {
    return <ManuscriptPage />;
  };
}
