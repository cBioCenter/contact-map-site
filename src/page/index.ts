export * from './ContactMapSitePage';
export * from './LandingPage';
export * from './ManuscriptPage';
