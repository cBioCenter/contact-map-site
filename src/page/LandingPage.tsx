import {
  BioblocksPDB,
  CONTACT_DISTANCE_PROXIMITY,
  CONTACT_MAP_DATA_TYPE,
  ContactMap,
  CouplingContainer,
  createContainerActions,
  createResiduePairActions,
  EMPTY_FUNCTION,
  generateResidueMapping,
  getCouplingScoresData,
  getPDBAndCouplingMismatch,
  IResidueMapping,
  IResidueMismatchResult,
  NGLContainer,
  PredictedContactMap,
  readFileAsText,
  VIZ_TYPE,
} from 'bioblocks-viz';
import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { Button, Grid, GridColumn, GridRow, Label, Message, Segment } from 'semantic-ui-react';

import { ErrorMessageComponent, FolderUploadComponent } from '~contact-map-site~/component';

export interface ILandingPageProps {
  style: Exclude<React.CSSProperties, 'height' | 'width'>;
  clearAllResidues(): void;
  clearAllSecondaryStructures(): void;
}

export interface ILandingPageState {
  [VIZ_TYPE.CONTACT_MAP]: CONTACT_MAP_DATA_TYPE;
  arePredictionsAvailable: boolean;
  errorMsg: string;
  filenames: Partial<{
    couplings: string;
    pdb: string;
    residue_mapper: string;
  }>;
  isLoading: boolean;
  measuredProximity: CONTACT_DISTANCE_PROXIMITY;
  mismatches: IResidueMismatchResult[];
  pdbData?: BioblocksPDB;
  residueMapping: IResidueMapping[];
}

export class LandingPageClass extends React.Component<ILandingPageProps, ILandingPageState> {
  public static defaultProps = {
    clearAllResidues: EMPTY_FUNCTION,
    clearAllSecondaryStructures: EMPTY_FUNCTION,
    style: {
      backgroundColor: '#ffffff',
    },
  };

  protected static initialState: ILandingPageState = {
    [VIZ_TYPE.CONTACT_MAP]: {
      couplingScores: new CouplingContainer(),
      pdbData: undefined,
      secondaryStructures: [],
    },
    arePredictionsAvailable: false,
    errorMsg: '',
    filenames: {},
    isLoading: false,
    measuredProximity: CONTACT_DISTANCE_PROXIMITY.CLOSEST,
    mismatches: [],
    pdbData: undefined,
    residueMapping: [],
  };

  public constructor(props: ILandingPageProps) {
    super(props);
    this.state = LandingPageClass.initialState;
  }

  public componentDidUpdate(prevProps: ILandingPageProps, prevState: ILandingPageState) {
    const { measuredProximity, pdbData } = this.state;
    const { couplingScores } = this.state[VIZ_TYPE.CONTACT_MAP];

    let errorMsg = '';

    let newMismatches = this.state.mismatches;

    if (
      pdbData &&
      (couplingScores !== prevState[VIZ_TYPE.CONTACT_MAP].couplingScores || pdbData !== prevState.pdbData)
    ) {
      newMismatches = getPDBAndCouplingMismatch(pdbData, couplingScores);

      if (newMismatches.length >= 1) {
        errorMsg = `Error details: ${newMismatches.length} mismatch(es) detected between coupling scores and PDB!\
        For example, residue number ${newMismatches[0].resno} is '${
          newMismatches[0].pdbAminoAcid.threeLetterCode
        }' in the PDB but '${newMismatches[0].couplingAminoAcid.threeLetterCode}' in the coupling scores file.`;
      }
    }

    if (pdbData && measuredProximity !== prevState.measuredProximity) {
      this.setState({
        [VIZ_TYPE.CONTACT_MAP]: {
          couplingScores: pdbData.amendPDBWithCouplingScores(couplingScores.rankedContacts, measuredProximity),
          pdbData: { known: this.state.pdbData },
          secondaryStructures: [],
        },
        errorMsg,
        mismatches: newMismatches,
      });
    } else if (errorMsg.length >= 1) {
      this.setState({
        errorMsg,
        mismatches: newMismatches,
      });
    }
  }

  public render({ style } = this.props, { errorMsg, mismatches, pdbData } = this.state) {
    return (
      <div id="BioblocksVizApp" style={{ ...style, height: '1000px' }}>
        <ErrorMessageComponent
          couplingScores={this.state[VIZ_TYPE.CONTACT_MAP].couplingScores}
          errorMsg={errorMsg}
          pdbData={pdbData}
          mismatches={mismatches}
        />
        {!pdbData && this.renderStartMessage()}
        <Segment attached={true} raised={true}>
          {this.renderCouplingComponents()}
        </Segment>
        {this.renderFooter()}
      </div>
    );
  }

  protected renderCouplingComponents = (
    { style } = this.props,
    { arePredictionsAvailable, measuredProximity, pdbData } = this.state,
  ) => (
    <Grid centered={true} padded={true} relaxed={true}>
      {this.renderUploadButtonsRow()}
      <Grid.Row>
        <br />
      </Grid.Row>
      <GridRow columns={2}>
        <GridColumn width={7}>{this.renderContactMapCard(arePredictionsAvailable, '500px', style, pdbData)}</GridColumn>
        <GridColumn width={7}>{this.renderNGLCard(measuredProximity, pdbData)}</GridColumn>
      </GridRow>
    </Grid>
  );

  protected renderFooter = () => {
    return (
      <footer style={{ padding: '25vh 0 25px 25px' }}>
        <>
          Powered by <a href="https://github.com/cBioCenter/bioblocks-viz">Bioblocks</a>!
        </>
      </footer>
    );
  };

  protected renderStartMessage = () => (
    <Message>
      {`To get started, please upload either a PDB (.pdb) or EVCouplings score (.csv) file!`} {<br />} Check out the
      {/* tslint:disable-next-line:no-http-string */}
      {<a href="http://evfold.org"> EVFold</a>}, {<a href="http://sanderlab.org/contact-maps/">Sander Lab</a>}, or
      {<a href="https://evcouplings.org/"> EVCouplings </a>} website to get these files.
    </Message>
  );

  protected renderUploadForm = (
    onChange: (e: React.ChangeEvent) => void,
    id: string,
    content: string,
    disabled?: boolean,
    accepts: string[] = [],
  ) => {
    return (
      <GridColumn>
        <Label as="label" basic={true} htmlFor={id}>
          <Button
            disabled={disabled}
            icon={'upload'}
            label={{
              basic: true,
              content,
            }}
            labelPosition={'right'}
          />
          <input
            accept={`.${accepts.join(',.')}`}
            disabled={disabled}
            id={id}
            onChange={onChange}
            hidden={true}
            type={'file'}
            required={true}
            multiple={false}
          />
        </Label>
      </GridColumn>
    );
  };

  protected renderContactMapCard = (
    arePredictionsAvailable: boolean,
    size: number | string,
    style: React.CSSProperties,
    pdbData?: BioblocksPDB,
  ) =>
    arePredictionsAvailable ? (
      <PredictedContactMap
        data={{
          couplingScores: this.state[VIZ_TYPE.CONTACT_MAP].couplingScores,
          pdbData: { known: pdbData },
          secondaryStructures: this.state[VIZ_TYPE.CONTACT_MAP].secondaryStructures,
        }}
        height={size}
        isDataLoading={this.state.isLoading}
        style={style}
        width={size}
      />
    ) : (
      <ContactMap
        data={{
          couplingScores: this.state[VIZ_TYPE.CONTACT_MAP].couplingScores,
          pdbData: { known: pdbData },
          secondaryStructures: this.state[VIZ_TYPE.CONTACT_MAP].secondaryStructures,
        }}
        height={size}
        isDataLoading={this.state.isLoading}
        style={style}
        width={size}
      />
    );

  protected renderNGLCard = (measuredProximity: CONTACT_DISTANCE_PROXIMITY, pdbData?: BioblocksPDB) => (
    <NGLContainer
      data={pdbData}
      isDataLoading={this.state.isLoading}
      measuredProximity={measuredProximity}
      onMeasuredProximityChange={this.onMeasuredProximityChange()}
    />
  );

  protected renderUploadButtonsRow = () => (
    <GridRow columns={4} textAlign={'center'} verticalAlign={'bottom'}>
      <GridColumn style={{ height: '100%' }}>
        <GridRow style={{ height: '100%' }}>
          <FolderUploadComponent onDrop={this.onFolderUpload} style={{ height: '100%' }} />
        </GridRow>
      </GridColumn>
      <GridColumn>{this.renderClearAllButton()}</GridColumn>
    </GridRow>
  );

  protected renderClearAllButton = () => (
    <GridRow verticalAlign={'middle'} columns={1} centered={true}>
      <GridColumn>
        <Label as="label" basic={true} htmlFor={'clear-data'}>
          <Button
            icon={'trash'}
            label={{
              basic: true,
              content: 'Clean View',
            }}
            labelPosition={'right'}
            onClick={this.onClearAll()}
          />
        </Label>
      </GridColumn>
    </GridRow>
  );

  protected onClearAll = () => async () => {
    const { clearAllResidues, clearAllSecondaryStructures } = this.props;
    this.setState(LandingPageClass.initialState);
    clearAllResidues();
    clearAllSecondaryStructures();
    this.forceUpdate();
  };

  protected onFolderUpload = async (files: File[]) => {
    await this.onClearAll()();
    const { measuredProximity } = this.state;
    this.setState({
      isLoading: true,
    });

    const filenames = {
      couplings: '',
      pdb: '',
      residue_mapper: '',
    };
    let couplingScoresCSV: string = '';
    let pdbData: BioblocksPDB = BioblocksPDB.createEmptyPDB();
    let residueMapping: IResidueMapping[] = [];

    for (const file of files) {
      if (file.name.endsWith('.pdb')) {
        pdbData = await BioblocksPDB.createPDB(file);
        filenames.pdb = file.name;
      } else {
        const parsedFile = await readFileAsText(file);
        if (
          file.name.endsWith('indextable') ||
          file.name.endsWith('indextableplus') ||
          file.name.startsWith('residue_mapping')
        ) {
          residueMapping = generateResidueMapping(parsedFile);
          filenames.residue_mapper = file.name;
        } else if (file.name.endsWith('.csv')) {
          couplingScoresCSV = parsedFile;
          filenames.couplings = file.name;
        }
      }
    }

    let couplingScores = getCouplingScoresData(couplingScoresCSV, residueMapping);
    couplingScores = pdbData.amendPDBWithCouplingScores(couplingScores.rankedContacts, measuredProximity);
    const mismatches = pdbData.getResidueNumberingMismatches(couplingScores);

    this.setState({
      [VIZ_TYPE.CONTACT_MAP]: {
        couplingScores,
        pdbData: { known: pdbData },
        secondaryStructures: pdbData.secondaryStructureSections,
      },
      arePredictionsAvailable: true,
      errorMsg: '',
      filenames,
      isLoading: false,
      mismatches,
      pdbData,
    });
  };

  protected onMeasuredProximityChange = () => (value: number) => {
    this.setState({
      measuredProximity: Object.values(CONTACT_DISTANCE_PROXIMITY)[value] as CONTACT_DISTANCE_PROXIMITY,
    });
  };
}

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      clearAllResidues: () => ({
        ...createResiduePairActions().candidates.clear(),
        ...createResiduePairActions().hovered.clear(),
        ...createResiduePairActions().locked.clear(),
      }),
      clearAllSecondaryStructures: () => ({
        ...createContainerActions('secondaryStructure/hovered').clear(),
        ...createContainerActions('secondaryStructure/selected').clear(),
      }),
    },
    dispatch,
  );

export const LandingPage = connect(
  null,
  mapDispatchToProps,
)(LandingPageClass);
