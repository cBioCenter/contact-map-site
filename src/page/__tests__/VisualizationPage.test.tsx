import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import { Modal } from 'semantic-ui-react';

import { BioblocksPDB, CONTACT_DISTANCE_PROXIMITY, Store } from 'bioblocks-viz';
import { FolderUploadComponent } from '~contact-map-site~/component';
import { VisualizationPageClass } from '~contact-map-site~/page';

describe('VisualizationPage', () => {
  beforeEach(() => {
    jest.resetModuleRegistry();
    jest.dontMock('ngl');
  });

  it('Should match existing snapshot when no props are provided.', () => {
    const wrapper = shallow(<VisualizationPageClass />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when hooked up to a Redux store.', () => {
    const wrapper = mount(
      <Provider store={Store}>
        <VisualizationPageClass />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when given a PDB.', async () => {
    const pdbData = await BioblocksPDB.createPDB('sample.pdb');
    const wrapper = shallow(<VisualizationPageClass />);
    wrapper.setState({
      pdbData,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when using closest proximity.', async () => {
    const pdbData = await BioblocksPDB.createPDB('sample.pdb');
    const wrapper = shallow(<VisualizationPageClass />);
    wrapper.setState({
      measuredProximity: CONTACT_DISTANCE_PROXIMITY.CLOSEST,
      pdbData,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when using c-alpha proximity.', async () => {
    const pdbData = await BioblocksPDB.createPDB('sample.pdb');
    const wrapper = shallow(<VisualizationPageClass />);
    wrapper.setState({
      measuredProximity: CONTACT_DISTANCE_PROXIMITY.C_ALPHA,
      pdbData,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update the state based on drag events.', () => {
    const wrapper = shallow(<VisualizationPageClass />);
    const instance = wrapper.instance() as VisualizationPageClass;
    expect(instance.state.isDragHappening).toBe(false);
    window.dispatchEvent(new Event('dragover'));
    expect(instance.state.isDragHappening).toBe(true);

    const onClose = wrapper.find(Modal).props().onClose;
    if (!onClose) {
      expect(onClose).not.toBeUndefined();
    } else {
      const event = { clientX: 100, clientY: 0 };
      onClose(event as React.MouseEvent<HTMLElement>, {});
      expect(instance.state.isDragHappening).toBe(false);
    }
  });

  it('Should handle files/folders being dragged.', async () => {
    const wrapper = shallow(<VisualizationPageClass />);
    const instance = wrapper.instance() as VisualizationPageClass;
    expect(instance.state.filenames).toEqual({});
    const onDrop = wrapper.find(FolderUploadComponent).props().onDrop;
    if (!onDrop) {
      expect(onDrop).not.toBeUndefined();
    } else {
      const event = new Event('drop');
      const couplingFile = new File([], 'coupling_scores.csv');
      const pdbFile = new File([], 'mock.pdb');
      const residueMapFile = new File([], 'residue_mapping.csv');

      await onDrop([pdbFile, residueMapFile, couplingFile], [], event);
      expect(instance.state.filenames).toEqual({
        couplings: 'coupling_scores.csv',
        pdb: 'mock.pdb',
        residue_mapper: 'residue_mapping.csv',
      });
    }
  });

  it('Should handle mismatches.', async () => {
    const wrapper = shallow(<VisualizationPageClass />);
    const instance = wrapper.instance() as VisualizationPageClass;
    expect(instance.state.mismatches).toEqual([]);
    const onDrop = wrapper.find(FolderUploadComponent).props().onDrop;
    if (!onDrop) {
      expect(onDrop).not.toBeUndefined();
    } else {
      const couplingScoresCsv =
        '145,81,0.79312,7.5652,A,A,0.9,2.4,47,1.0,E,R\n\
      179,66,0.78681,3.5872,A,A,0.9,1.3,37,1.0,T,M';

      const event = new Event('drop');
      const couplingFile = new File([couplingScoresCsv], 'coupling_scores.csv');
      const pdbFile = new File([], 'mock.pdb');
      const residueMapFile = new File([], 'residue_mapping.csv');

      await onDrop([pdbFile, residueMapFile, couplingFile], [], event);
      expect(instance.state.mismatches).not.toEqual([]);
    }
  });
});
