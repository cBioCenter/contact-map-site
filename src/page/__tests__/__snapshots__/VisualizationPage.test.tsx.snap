// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualizationPage Should match existing snapshot when given a PDB. 1`] = `
<div
  id="BioblocksVizApp"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "1000px",
    }
  }
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    closeOnEscape={true}
    closeOnPortalMouseLeave={true}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="fullscreen"
  >
    <FolderUploadComponent
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "border": "7px dashed blue",
          "height": "90vmin",
          "textAlign": "center",
        }
      }
    />
  </Modal>
  <ErrorMessageComponent
    couplingScores={
      e {
        "colCounter": 0,
        "contacts": Array [],
        "derivedFromCouplingScoresFlag": true,
        "indexRange": Object {
          "max": 1,
          "min": 1,
        },
        "rowCounter": 0,
        "scoreSource": "cn",
        "totalStoredContacts": 0,
      }
    }
    errorMsg=""
    mismatches={Array []}
  />
  <Message>
    To get started, please drag and drop onto the page either:
    <br />
    (1) An evcouplings results directory.
    <br />
    (2) Individual .pdb, coupling scores
    <Popup
      content="A file that ends as 'CouplingScores.csv'"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    , and residue mapping files
    <Popup
      content="A .csv file that starts with 'residue_mapping' - or - A file that ends in .indextable / .indextableplus"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    .
    <br />
    <br />
     Check out the
    <a
      href="http://evfold.org"
    >
       EVFold
    </a>
     or
    <a
      href="https://evcouplings.org/"
    >
       EVCouplings 
    </a>
     website to get these files.
  </Message>
  <Segment
    attached={true}
    raised={true}
  >
    <Grid
      centered={true}
      padded={true}
      relaxed={true}
    >
      <GridRow
        textAlign="right"
        verticalAlign="bottom"
      >
        <GridColumn
          style={
            Object {
              "height": "100%",
              "width": "auto",
            }
          }
        >
          <Button
            as="button"
            icon="trash"
            label={
              Object {
                "content": "Clean View",
              }
            }
            labelPosition="right"
            onClick={[Function]}
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </GridColumn>
      </GridRow>
      <GridRow>
        <br />
      </GridRow>
      <GridRow
        columns={2}
        verticalAlign="bottom"
      >
        <GridColumn
          width={7}
        >
          <Component
            data={
              Object {
                "couplingScores": e {
                  "colCounter": 0,
                  "contacts": Array [],
                  "derivedFromCouplingScoresFlag": true,
                  "indexRange": Object {
                    "max": 1,
                    "min": 1,
                  },
                  "rowCounter": 0,
                  "scoreSource": "cn",
                  "totalStoredContacts": 0,
                },
                "pdbData": Object {
                  "experimental": undefined,
                },
                "secondaryStructures": Array [],
              }
            }
            isDataLoading={false}
            style={
              Object {
                "backgroundColor": "#ffffff",
              }
            }
          />
        </GridColumn>
        <GridColumn
          width={7}
        >
          <Component
            experimentalProteins={Array []}
            isDataLoading={false}
            measuredProximity="CLOSEST ATOM"
            onMeasuredProximityChange={[Function]}
            predictedProteins={Array []}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </Segment>
  <footer
    style={
      Object {
        "padding": "25vh 0 25px 25px",
      }
    }
  >
    Powered by 
    <a
      href="https://github.com/cBioCenter/bioblocks-viz"
    >
      Bioblocks
    </a>
    !
  </footer>
</div>
`;

exports[`VisualizationPage Should match existing snapshot when hooked up to a Redux store. 1`] = `
<h
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <withBBStore(Connect(VisualizationPageClass)) />
</h>
`;

exports[`VisualizationPage Should match existing snapshot when no props are provided. 1`] = `
<div
  id="BioblocksVizApp"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "1000px",
    }
  }
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    closeOnEscape={true}
    closeOnPortalMouseLeave={true}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="fullscreen"
  >
    <FolderUploadComponent
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "border": "7px dashed blue",
          "height": "90vmin",
          "textAlign": "center",
        }
      }
    />
  </Modal>
  <ErrorMessageComponent
    couplingScores={
      e {
        "colCounter": 0,
        "contacts": Array [],
        "derivedFromCouplingScoresFlag": true,
        "indexRange": Object {
          "max": 1,
          "min": 1,
        },
        "rowCounter": 0,
        "scoreSource": "cn",
        "totalStoredContacts": 0,
      }
    }
    errorMsg=""
    mismatches={Array []}
  />
  <Message>
    To get started, please drag and drop onto the page either:
    <br />
    (1) An evcouplings results directory.
    <br />
    (2) Individual .pdb, coupling scores
    <Popup
      content="A file that ends as 'CouplingScores.csv'"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    , and residue mapping files
    <Popup
      content="A .csv file that starts with 'residue_mapping' - or - A file that ends in .indextable / .indextableplus"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    .
    <br />
    <br />
     Check out the
    <a
      href="http://evfold.org"
    >
       EVFold
    </a>
     or
    <a
      href="https://evcouplings.org/"
    >
       EVCouplings 
    </a>
     website to get these files.
  </Message>
  <Segment
    attached={true}
    raised={true}
  >
    <Grid
      centered={true}
      padded={true}
      relaxed={true}
    >
      <GridRow
        textAlign="right"
        verticalAlign="bottom"
      >
        <GridColumn
          style={
            Object {
              "height": "100%",
              "width": "auto",
            }
          }
        >
          <Button
            as="button"
            icon="trash"
            label={
              Object {
                "content": "Clean View",
              }
            }
            labelPosition="right"
            onClick={[Function]}
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </GridColumn>
      </GridRow>
      <GridRow>
        <br />
      </GridRow>
      <GridRow
        columns={2}
        verticalAlign="bottom"
      >
        <GridColumn
          width={7}
        >
          <Component
            data={
              Object {
                "couplingScores": e {
                  "colCounter": 0,
                  "contacts": Array [],
                  "derivedFromCouplingScoresFlag": true,
                  "indexRange": Object {
                    "max": 1,
                    "min": 1,
                  },
                  "rowCounter": 0,
                  "scoreSource": "cn",
                  "totalStoredContacts": 0,
                },
                "pdbData": Object {
                  "experimental": undefined,
                },
                "secondaryStructures": Array [],
              }
            }
            isDataLoading={false}
            style={
              Object {
                "backgroundColor": "#ffffff",
              }
            }
          />
        </GridColumn>
        <GridColumn
          width={7}
        >
          <Component
            experimentalProteins={Array []}
            isDataLoading={false}
            measuredProximity="CLOSEST ATOM"
            onMeasuredProximityChange={[Function]}
            predictedProteins={Array []}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </Segment>
  <footer
    style={
      Object {
        "padding": "25vh 0 25px 25px",
      }
    }
  >
    Powered by 
    <a
      href="https://github.com/cBioCenter/bioblocks-viz"
    >
      Bioblocks
    </a>
    !
  </footer>
</div>
`;

exports[`VisualizationPage Should match existing snapshot when using c-alpha proximity. 1`] = `
<div
  id="BioblocksVizApp"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "1000px",
    }
  }
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    closeOnEscape={true}
    closeOnPortalMouseLeave={true}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="fullscreen"
  >
    <FolderUploadComponent
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "border": "7px dashed blue",
          "height": "90vmin",
          "textAlign": "center",
        }
      }
    />
  </Modal>
  <ErrorMessageComponent
    couplingScores={
      e {
        "colCounter": 0,
        "contacts": Array [],
        "derivedFromCouplingScoresFlag": true,
        "indexRange": Object {
          "max": 1,
          "min": 1,
        },
        "rowCounter": 0,
        "scoreSource": "cn",
        "totalStoredContacts": 0,
      }
    }
    errorMsg=""
    mismatches={Array []}
  />
  <Message>
    To get started, please drag and drop onto the page either:
    <br />
    (1) An evcouplings results directory.
    <br />
    (2) Individual .pdb, coupling scores
    <Popup
      content="A file that ends as 'CouplingScores.csv'"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    , and residue mapping files
    <Popup
      content="A .csv file that starts with 'residue_mapping' - or - A file that ends in .indextable / .indextableplus"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    .
    <br />
    <br />
     Check out the
    <a
      href="http://evfold.org"
    >
       EVFold
    </a>
     or
    <a
      href="https://evcouplings.org/"
    >
       EVCouplings 
    </a>
     website to get these files.
  </Message>
  <Segment
    attached={true}
    raised={true}
  >
    <Grid
      centered={true}
      padded={true}
      relaxed={true}
    >
      <GridRow
        textAlign="right"
        verticalAlign="bottom"
      >
        <GridColumn
          style={
            Object {
              "height": "100%",
              "width": "auto",
            }
          }
        >
          <Button
            as="button"
            icon="trash"
            label={
              Object {
                "content": "Clean View",
              }
            }
            labelPosition="right"
            onClick={[Function]}
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </GridColumn>
      </GridRow>
      <GridRow>
        <br />
      </GridRow>
      <GridRow
        columns={2}
        verticalAlign="bottom"
      >
        <GridColumn
          width={7}
        >
          <Component
            data={
              Object {
                "couplingScores": e {
                  "colCounter": 0,
                  "contacts": Array [],
                  "derivedFromCouplingScoresFlag": true,
                  "indexRange": Object {
                    "max": 1,
                    "min": 1,
                  },
                  "rowCounter": 0,
                  "scoreSource": "cn",
                  "totalStoredContacts": 0,
                },
                "pdbData": Object {
                  "experimental": undefined,
                },
                "secondaryStructures": Array [],
              }
            }
            isDataLoading={false}
            style={
              Object {
                "backgroundColor": "#ffffff",
              }
            }
          />
        </GridColumn>
        <GridColumn
          width={7}
        >
          <Component
            experimentalProteins={Array []}
            isDataLoading={false}
            measuredProximity="C-α"
            onMeasuredProximityChange={[Function]}
            predictedProteins={Array []}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </Segment>
  <footer
    style={
      Object {
        "padding": "25vh 0 25px 25px",
      }
    }
  >
    Powered by 
    <a
      href="https://github.com/cBioCenter/bioblocks-viz"
    >
      Bioblocks
    </a>
    !
  </footer>
</div>
`;

exports[`VisualizationPage Should match existing snapshot when using closest proximity. 1`] = `
<div
  id="BioblocksVizApp"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "1000px",
    }
  }
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    closeOnEscape={true}
    closeOnPortalMouseLeave={true}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="fullscreen"
  >
    <FolderUploadComponent
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "border": "7px dashed blue",
          "height": "90vmin",
          "textAlign": "center",
        }
      }
    />
  </Modal>
  <ErrorMessageComponent
    couplingScores={
      e {
        "colCounter": 0,
        "contacts": Array [],
        "derivedFromCouplingScoresFlag": true,
        "indexRange": Object {
          "max": 1,
          "min": 1,
        },
        "rowCounter": 0,
        "scoreSource": "cn",
        "totalStoredContacts": 0,
      }
    }
    errorMsg=""
    mismatches={Array []}
  />
  <Message>
    To get started, please drag and drop onto the page either:
    <br />
    (1) An evcouplings results directory.
    <br />
    (2) Individual .pdb, coupling scores
    <Popup
      content="A file that ends as 'CouplingScores.csv'"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    , and residue mapping files
    <Popup
      content="A .csv file that starts with 'residue_mapping' - or - A file that ends in .indextable / .indextableplus"
      disabled={false}
      eventsEnabled={true}
      offset={0}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      position="top left"
      trigger={
        <a>
          (?)
        </a>
      }
    />
    .
    <br />
    <br />
     Check out the
    <a
      href="http://evfold.org"
    >
       EVFold
    </a>
     or
    <a
      href="https://evcouplings.org/"
    >
       EVCouplings 
    </a>
     website to get these files.
  </Message>
  <Segment
    attached={true}
    raised={true}
  >
    <Grid
      centered={true}
      padded={true}
      relaxed={true}
    >
      <GridRow
        textAlign="right"
        verticalAlign="bottom"
      >
        <GridColumn
          style={
            Object {
              "height": "100%",
              "width": "auto",
            }
          }
        >
          <Button
            as="button"
            icon="trash"
            label={
              Object {
                "content": "Clean View",
              }
            }
            labelPosition="right"
            onClick={[Function]}
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </GridColumn>
      </GridRow>
      <GridRow>
        <br />
      </GridRow>
      <GridRow
        columns={2}
        verticalAlign="bottom"
      >
        <GridColumn
          width={7}
        >
          <Component
            data={
              Object {
                "couplingScores": e {
                  "colCounter": 0,
                  "contacts": Array [],
                  "derivedFromCouplingScoresFlag": true,
                  "indexRange": Object {
                    "max": 1,
                    "min": 1,
                  },
                  "rowCounter": 0,
                  "scoreSource": "cn",
                  "totalStoredContacts": 0,
                },
                "pdbData": Object {
                  "experimental": undefined,
                },
                "secondaryStructures": Array [],
              }
            }
            isDataLoading={false}
            style={
              Object {
                "backgroundColor": "#ffffff",
              }
            }
          />
        </GridColumn>
        <GridColumn
          width={7}
        >
          <Component
            experimentalProteins={Array []}
            isDataLoading={false}
            measuredProximity="CLOSEST ATOM"
            onMeasuredProximityChange={[Function]}
            predictedProteins={Array []}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </Segment>
  <footer
    style={
      Object {
        "padding": "25vh 0 25px 25px",
      }
    }
  >
    Powered by 
    <a
      href="https://github.com/cBioCenter/bioblocks-viz"
    >
      Bioblocks
    </a>
    !
  </footer>
</div>
`;
